atomic_reactor/util.py

```diff
@@ -652,13 +660,25 @@ def __init__(self, registry, insecure=False, dockercfg_path=None):
         self.insecure = insecure
 
         self.auth = None
+        self.cert_path = None
+        self.key_path = None
         if dockercfg_path:
-            dockercfg = Dockercfg(dockercfg_path).get_credentials(registry)
+            dockercfg = Dockercfg(dockercfg_path)
+            if dockercfg.json_secret_path is not None:
+                dockercfg = dockercfg.get_credentials(registry)
+
+                username = dockercfg.get('username')
+                password = dockercfg.get('password')
+                if username and password:
+                    self.auth = requests.auth.HTTPBasicAuth(username, password)
+            else:
+                # Use the registry certificates to authenticate instead of
+                # instead of username/password.
+                self.cert_path = os.path.join(dockercfg_path, 'registry.crt')
+                self.key_path = os.path.join(dockercfg_path, 'registry.key')
```

Comment:
Oh! I had thought there was some way a .docker/config.json (= 'dockercfg' secret type) could hold a certificate and key, but it seems not. In that  case, this seems to be an ad-hoc method for storing a certificate and key. :(

Ideally we'd use the same mechanism that e.g. Kubernetes and/or Docker uses for this.

For Docker, it seems that /etc/docker/certs.d/ is used, and the filenames there (per registry) are client.cert, client.key, and ca.crt (yes, .crt in one filename and .cert in another).

At the very least I'd hope we'd use the same filenames Docker does.

Does Kubernetes do the same thing? I can't seem to find any documentation about it. In fact, there's very little documentation anyway I can find about securing docker registries with certificates instead of basic auth.

Reply:
> At the very least I'd hope we'd use the same filenames Docker does.

Ok, I will update the code to use client.crt and client.key.

>Does Kubernetes do the same thing? I can't seem to find any documentation about it. In fact, there's very little documentation anyway I can find about securing docker registries with certificates instead of basic auth.

I  don't have much history on why we secured the registry that way in Fedora. Maybe @maxamillion or @puiterwijk could help with that